webapp/src/lib/components/spot-listings/spots-list-component.svelte

```diff
@@ -0,0 +1,81 @@
+<script lang="ts">
+    import { ChargingStation, PlugFilled, Home } from 'carbon-icons-svelte';
+    import { TooltipIcon } from 'carbon-components-svelte';
+    import DetailModal from './detail-modal.svelte';
+    export let listing;
+
+    let modalOpen = false;
+
+    const handleClick = () => {
+        modalOpen = true;
+    };
+</script>
+
+<div
+    class="listing-info"
+    on:click={() => handleClick()}
+    on:keyup={() => {}}
+    role="button"
+    tabindex="0"
+>
+    <div class="address">
+        <p class="listing-head">{listing.location.street_address}</p>
+        <p>{listing.location.postal_code}</p>
+    </div>
+    <div class="features">
+        <div class="feature">
+            {#if listing.features.charging_station}
+                <TooltipIcon tooltipText="Charging Station Available" direction="right" align="end">
+                    <i><ChargingStation size={20} /></i>
+                </TooltipIcon>
+            {/if}
+        </div>
+        <div class="feature">
+            {#if listing.features.plug_in}
+                <TooltipIcon tooltipText="Plug-in Available" direction="right" align="end">
+                    <i class="icon"><PlugFilled size={20} /></i>
+                </TooltipIcon>
+            {/if}
+        </div>
+        <div class="feature">
+            {#if listing.features.shelter}
+                <TooltipIcon tooltipText="Shelter Available" direction="right" align="end">
+                    <i class="icon"><Home size={20} /></i>
+                </TooltipIcon>
+            {/if}
+        </div>
+    </div>
+</div>
+<DetailModal bind:open={modalOpen} {listing} />
```

Comment:
Please move this DetailModal to +page.svelte under routes/app. If you add it here then you end up with multiple Modal invisible Modal for each listing.

Reply:
Why would that happen, I am not sure I understand

Reply:
the Detail Modal is spawned on screen by default whether it is close or not. Closing a Modal only means it is invisible. So if you put it here, you will have an active Modal for every spot listing you have.